### PR TITLE
Add v0.44.0 to releases.md

### PR DESCRIPTION
**Pull request by lbernick**
# Changes

Update releases.
/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- n/a Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- n/a Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- n/a Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: releases.md [hunk 1]
```diff
@@ -39,7 +39,14 @@ Further documentation available:
...
 
-## Releases
+## Release
+
+### v0.44 (LTS)
+
+- **Latest Release**: [v0.44.0][v0-44-0] (2023-01-24) ([docs][v0-44-0-docs], [examples][v0-44-0-examples])
+- **Initial Release**: [v0.44.0][v0-44-0] (2023-01-24)
+- **End of Life**: 2024-01-24
+- **Patch Releases**: [v0.40.0][v0-40-0]
```
Comment (pritidesai) [inline releases.md:49]: Should this be `[v0.44.0][v0-44-0]`?

Comment (lbernick) [reply releases.md]: whoops! fixed, thanks

**Review by pritidesai — APPROVED** [review]
thanks @lbernick 🎉

**Pull request merged**